src/machine/machine.go

```diff
@@ -15,3 +15,7 @@ func (p GPIO) High() {
 func (p GPIO) Low() {
 	p.Set(false)
 }
+
+type PWM struct {
+	Pin uint8
```

Comment:
Is there a reason not to re-use the `GPIO` object? Perhaps it can be renamed to `Pin` if it supports more than just setting pins.

Reply:
Well, actually, I was windering if perhaps it might be a good idea to create a GPIO interface, and separately a PWM interface, and then have a specific machine implement whatever interface(s) for a pin on that platform. This would help to ensure some consistency across platforms. WHat do you think?

Reply:
I don't think I understand the need for interfaces here? It's nice to have a similar set of methods as other APIs (for duck typing) but otherwise I don't see the need. TinyGo programs are compiled for one specific chip so can simply use a `machine.GPIO` (for example) without having to care too much how that is implemented. This doesn't need an interface, just a similar set of methods across platforms.

Aside from that, I'm a little bit cautious with interfaces, as they can cause code bloat. I've done some optimizations that reduce the overhead by a large margin but still they aren't free so should be used only when necessary.

Reply:
Having the appropriate `Configure()` for PWM pins vs. GPIO pins seems to me like a nicer API. We can still avoid the interface for the moment.

Reply:
Ah, right. `Configure()` (and `Set()`). Then I don't see an obvious solution. Let's keep it as is for now.